src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/readwrite/SetReadWriteStep.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.step.readwrite
+
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.elasticapi.suspendUntil
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.ManagedIndexMetaData
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.ReadWriteActionConfig
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.step.Step
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.action.admin.indices.settings.put.UpdateSettingsRequest
+import org.elasticsearch.action.support.master.AcknowledgedResponse
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.service.ClusterService
+
+class SetReadWriteStep(
+    val clusterService: ClusterService,
+    val client: Client,
+    val config: ReadWriteActionConfig,
+    managedIndexMetaData: ManagedIndexMetaData
+) : Step("set_read_write", managedIndexMetaData) {
+
+    private val logger = LogManager.getLogger(javaClass)
+    private var failed: Boolean = false
+    private var info: Map<String, Any>? = null
+
+    // TODO: Incorporate retries from config and consumed retries from metadata
+    override suspend fun execute() {
+        val updateSettingsRequest = UpdateSettingsRequest()
+            .indices(managedIndexMetaData.index)
+            .settings(
+                mapOf("index.blocks.write" to false)
+            )
+        val response: AcknowledgedResponse = client.admin().indices()
+            .suspendUntil { updateSettings(updateSettingsRequest, it) }
+
+        if (!response.isAcknowledged) {
+            failed = true
+            info = mapOf("message" to "Failed to set index to read-write")
+        }
+    }
+
+    override fun getUpdatedManagedIndexMetaData(currentMetaData: ManagedIndexMetaData): ManagedIndexMetaData {
+        return currentMetaData.copy(
+            step = name,
+            stepStartTime = getStepStartTime().toEpochMilli(),
+            transitionTo = null,
+            stepCompleted = !failed,
+            failed = failed,
+            info = info
```

Comment:
If we didn't fail, let's set the "info" to a successful message saying that the index was set to read-write since. We should always have a message in info to show on the UI/API after something is done.

Reply:
Good catch, added update to ```info``` when successful. 

Note: The not ```isAcknowledged``` cases for the time being have simple messages similar to the case with an ```Exception``` without the ```cause```. Let me know if you have any suggestions for more descriptive messages here.